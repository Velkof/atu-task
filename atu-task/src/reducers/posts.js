import { 
    GET_POSTS_REQUEST, 
    GET_POSTS_SUCCESS,
    GET_POSTS_FAIL,
    GET_POST_DETAILS_REQUEST,
    GET_POST_DETAILS_SUCCESS,
    GET_POST_AUTHOR_SUCCESS,
    GET_POST_AUTHOR_FAIL
} from '../constants/index';

export const initialState = {
    loading: true,
    posts: [],
    selectedPost: {},
    selectedPostAuthor: {}
}

export const posts = (state = initialState, action) => {
    switch(action.type) {
        case GET_POSTS_REQUEST:
            return { ...state, loading: true }
        case GET_POSTS_SUCCESS:
            return { ...state, loading: false, posts: action.payload}
        case GET_POSTS_FAIL:
            return { ...state, loading: false }    
        case GET_POST_DETAILS_REQUEST:
            return { ...state, loading: true }    
        case GET_POST_DETAILS_SUCCESS:
                return { ...state, selectedPost: action.payload}
        case GET_POST_AUTHOR_SUCCESS:
            return { ...state, loading: false, selectedPostAuthor: action.payload}
        case GET_POST_AUTHOR_FAIL:
            return { ...state, loading: false }                            
        default:
            return state;
    }
}

export default posts
  