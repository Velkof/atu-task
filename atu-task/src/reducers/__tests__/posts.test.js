import reducer from '../posts'
import { initialState } from '../posts'
import { 
    GET_POSTS_REQUEST, 
    GET_POSTS_SUCCESS,
    GET_POSTS_FAIL,
    GET_POST_DETAILS_REQUEST,
    GET_POST_DETAILS_SUCCESS,
    GET_POST_AUTHOR_SUCCESS,
    GET_POST_AUTHOR_FAIL
} from '../../constants/index';

const post = {
    id: 1, 
    userId: 1, 
    title: 'Post title', 
    body: 'Post body'
}

const user = {
    id: 1, 
    name: 'User name' 
}

describe('Returns the correct state when', () => {
    it('it is INITIAL_STATE', () => {
      expect(reducer(undefined, {})).toMatchSnapshot()
    })

    it('it receives GET_POSTS_REQUEST', () => {
        const action = { 
            type: GET_POSTS_REQUEST
        };
        expect(reducer(initialState, action)).toMatchSnapshot()
    });

    it('it receives GET_POSTS_SUCCESS', () => {
        const action = { 
            type: GET_POSTS_SUCCESS, 
            payload: { posts: [ post ] }
        };
        expect(reducer(initialState, action)).toMatchSnapshot()
    });

    it('it receives GET_POSTS_FAIL', () => {
        const action = { 
            type: GET_POSTS_FAIL
        };
        expect(reducer(initialState, action)).toMatchSnapshot()
    });

    it('it receives GET_POST_DETAILS_REQUEST', () => {
        const action = { 
            type: GET_POST_DETAILS_REQUEST
        };
        expect(reducer(initialState, action)).toMatchSnapshot()
    });

    it('it receives GET_POST_DETAILS_SUCCESS', () => {
        const action = { 
            type: GET_POST_DETAILS_SUCCESS, 
            payload: { post: post }
        };
        expect(reducer(initialState, action)).toMatchSnapshot()
    });

    it('it receives GET_POST_AUTHOR_SUCCESS', () => {
        const action = { 
            type: GET_POST_AUTHOR_SUCCESS, 
            payload: { user: user }
        };
        expect(reducer(initialState, action)).toMatchSnapshot()
    });

    it('it receives GET_POST_AUTHOR_FAIL', () => {
        const action = { 
            type: GET_POST_AUTHOR_FAIL
        };
        expect(reducer(initialState, action)).toMatchSnapshot()
    });
})

