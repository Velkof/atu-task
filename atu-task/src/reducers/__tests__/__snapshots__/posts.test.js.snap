// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Returns the correct state when it is INITIAL_STATE 1`] = `
Object {
  "loading": true,
  "posts": Array [],
  "selectedPost": Object {},
  "selectedPostAuthor": Object {},
}
`;

exports[`Returns the correct state when it receives GET_POST_AUTHOR_FAIL 1`] = `
Object {
  "loading": false,
  "posts": Array [],
  "selectedPost": Object {},
  "selectedPostAuthor": Object {},
}
`;

exports[`Returns the correct state when it receives GET_POST_AUTHOR_SUCCESS 1`] = `
Object {
  "loading": false,
  "posts": Array [],
  "selectedPost": Object {},
  "selectedPostAuthor": Object {
    "user": Object {
      "id": 1,
      "name": "User name",
    },
  },
}
`;

exports[`Returns the correct state when it receives GET_POST_DETAILS_REQUEST 1`] = `
Object {
  "loading": true,
  "posts": Array [],
  "selectedPost": Object {},
  "selectedPostAuthor": Object {},
}
`;

exports[`Returns the correct state when it receives GET_POST_DETAILS_SUCCESS 1`] = `
Object {
  "loading": true,
  "posts": Array [],
  "selectedPost": Object {
    "post": Object {
      "body": "Post body",
      "id": 1,
      "title": "Post title",
      "userId": 1,
    },
  },
  "selectedPostAuthor": Object {},
}
`;

exports[`Returns the correct state when it receives GET_POSTS_FAIL 1`] = `
Object {
  "loading": false,
  "posts": Array [],
  "selectedPost": Object {},
  "selectedPostAuthor": Object {},
}
`;

exports[`Returns the correct state when it receives GET_POSTS_REQUEST 1`] = `
Object {
  "loading": true,
  "posts": Array [],
  "selectedPost": Object {},
  "selectedPostAuthor": Object {},
}
`;

exports[`Returns the correct state when it receives GET_POSTS_SUCCESS 1`] = `
Object {
  "loading": false,
  "posts": Object {
    "posts": Array [
      Object {
        "body": "Post body",
        "id": 1,
        "title": "Post title",
        "userId": 1,
      },
    ],
  },
  "selectedPost": Object {},
  "selectedPostAuthor": Object {},
}
`;
